src/lib/components/setlist/CustomModal.svelte

```diff
@@ -0,0 +1,26 @@
+<script lang="ts">
+	import { Modal } from 'carbon-components-svelte'
+
+	export let open = false
+	export let onClose: () => void
+	export let onSubmit: () => Promise<boolean> // returns whether the function was executed successfully
+
+	/** Closes the modal and waits a short time to call the onClose to account for the closing animation of the modal. */
+	function closeModal() {
+		open = false
+		setTimeout(onClose, 250)
+	}
```

Comment:
Ok dan snap ik het dus toch niet, waarom is dit nodig? je kunt toch prima doorgaan met andere dingen terwijl de modal aan het sluiten is

Reply:
De `open = false` triggert de closing animation van de modal. De `onClose` method die meegegeven wordt zet de `modalState` in de parent component leeft op `{ state: "closed" }`. De modal wordt conditionally gerendered (`if modalState.state === 'MODAL_NAME`), dus als je onClose aanroept op hetzelfde moment als de closing animation begint, dan verdwijnt de modal altogether en wordt de animation dus niet afgespeeld. De timeout zorgt ervoor dat de animation eerst afgespeeld kan worden, waarna de modal niet meer gerenderd wordt als ie toch al niet meer zichtbaar is.